lib/SPIRV/CMakeLists.txt

```diff
@@ -45,12 +45,14 @@ add_llvm_library(LLVMSPIRVLib
     TransformUtils
   DEPENDS
     intrinsics_gen
+    spirv-headers
 )
 
 target_include_directories(LLVMSPIRVLib
   PRIVATE
     ${LLVM_INCLUDE_DIRS}
     ${LLVM_SPIRV_INCLUDE_DIRS}
+    ${LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR}/include/spirv/unified1
```

Comment:
From what I see, `SPIRV-Headers` are [defined](https://github.com/KhronosGroup/SPIRV-Headers/blob/master/CMakeLists.txt#L72) as [interface library](https://cmake.org/cmake/help/v3.4/command/add_library.html#interface-libraries), which simplifies its usage.

 If I understand correctly (educated myself using [this](http://mariobadr.com/creating-a-header-only-library-with-cmake.html) material), the only thing we need is to "link" the translator with SPIR-V headers:
 ```
target_link_libraries(LLVMSPIRVLib SPIRV-Headers)
# or probably add SPIRV-Headers as an argument to add_llvm_library above
 ```
 
 And since `SPIRV-Headers` is an interface library, we will inherit `target_include_directories` from it, which are defined [here](https://github.com/KhronosGroup/SPIRV-Headers/blob/master/CMakeLists.txt#L73).
 
 I don't have experience with interface libraries, but if we would be able to simplify this code, that would be great

Reply:
At last I have succeeded in implementing this interface library approach locally. The only issue I'm facing with finalizing it is that SPIRV-Headers exports the `include/` directory, and `spirv.hpp` lies in `include/spirv/unified1`. Using triangle brackets for `spirv.hpp` inclusion within the translator sources doesn't seem to work, so one of the following edits is required:
1. Export `include/spirv/unified1` as well (needs a patch to the SPIRV-Headers repo)
2. Use `#include "spirv/unified1/spirv.hpp"` in translator sources.
@svenvh, would it be acceptable if I went with **2** as a temporary solution and re-designed this in favor of option **1** as a follow-up?

Reply:
> @svenvh, would it be acceptable if I went with 2 as a temporary solution and re-designed this in favor of option 1 as a follow-up?

No objections from me.  I probably wouldn't even mind if 2 becomes a permanent solution, as it makes it clearer which spirv.hpp we're actually including.

Reply:
There are exporting issues for the `llvm/projects/SPIRV-Headers` approach when using the header-only INTERFACE "linking". I wasn't able to resolve it without touching LLVM/SPIRV-Headers build scripts, so I've decided to go with the earlier directory inclusion approach, while retaining configure-time download and adding a TODO for switching to the INTERFACE library approach.